analysis_runner/cli.py

```diff
@@ -153,7 +132,64 @@ def _get_google_auth_token() -> str:
     return creds.id_token
 
 
+def parse_args(args=None):
+    """
+    Parse args using argparse (
+    """
+    parser = argparse.ArgumentParser()
+    # https://docs.python.org/dev/library/argparse.html#action
+    parser.add_argument(
+        '-v',
+        '--version',
+        action='version',
+        version=f'analysis-runner v{_version.__version__}',
+    )
+    parser.add_argument(
+        '--dataset',
+        required=True,
+        type=str,
+        help='The dataset name, which determines which analysis-runner '
+        'server to send the request to.',
+    )
+    parser.add_argument(
+        '-o',
+        '--output-dir',
+        required=True,
+        type=str,
+        help='The output directory of the run, MUST start with "gs://".',
+    )
+    parser.add_argument(
+        '--repository',
+        '--repo',
+        help='The URI of the repository to run, must be approved by the appropriate '
+        'server. Default behavior is to find the repository of the current working '
+        'directory with `git remote get-url origin`.',
+    )
+    parser.add_argument(
+        '--commit',
+        help='The commit HASH or TAG of a commit to run, the default behavior is to '
+        'use the current commit of the local repository, however the literal value '
+        '"HEAD" is not allowed.',
+    )
+
+    parser.add_argument(
+        '--description',
+        required=True,
+        help='Human-readable description of the job, '
+        'logged together with the output data.',
+    )
+
+    parser.add_argument(
+        '--access-level',
+        choices=(['test', 'standard', 'full']),
+        default='test',
+        help='Which permissions to grant when running the job.',
+    )
+
+    parser.add_argument('script', nargs=argparse.REMAINDER, default=[])
+
+    return parser.parse_args(args)
+
+
 if __name__ == '__main__':
-    # Disable pylint because click decorates the function in a specific way
-    # pylint: disable=no-value-for-parameter
-    main()
+    main_from_args()
```

Comment:
Shouldn't this be `main_from_args(sys.argv)` or similar? Where do the `args` come from? :)

Reply:
Argparse automatically pulls from `sys.argv`.

Reply:
Got it, thanks!

Reply:
I added a comment to clarify this in code.